### PR TITLE
Rename llvm::ThreadPool -> llvm::DefaultThreadPool (NFC)

### PR DESCRIPTION
**Pull request by joker-eph**
The base class llvm::ThreadPoolInterface will be renamed llvm::ThreadPool
in a subsequent commit.

This is a breaking change: clients who use to create a ThreadPool must now
create a DefaultThreadPool.

### CODE REVIEW

**Review by kuhar — APPROVED** [review]
Makes sense to me

**Review by aganea — APPROVED** [review]

**Comment by joker-eph** [comment]
@dwblaikie : how would you split it? I didn't quite get the two renamings you have in mind?

**Comment by dwblaikie** [comment]
> @dwblaikie : how would you split it? I didn't quite get the two renamings you have in mind?

One patch `ThreadPool->DefaultThreadPool` (people get a build error about `ThreadPool` not being the name of anything, find this patch as the root cause, and rename all their ThreadPool->DefaultThreadPool)
Follow-up patch, `ThreadPoolInterface->ThreadPool` (similarly clear/separate errors)

Changing both in one patch risks folks getting confusing error messages because their existing ThreadPool usage will now instantly start being interpreted as the interface type - resulting in different/confusing errors about inability to instantiate abstract types, etc, presumably. Rather than just that the name is no longer present at all.

Ultimately they can root cause and figure out both renamings - probably not a huge deal either way, but my understanding was that separating them might be marginally better for downstrteamers.

**Comment by pogo59** [comment]
> separating them might be marginally better for downstrteamers.

Hear, hear.

**Comment by joker-eph** [comment]
> One patch ThreadPool->DefaultThreadPool (people get a build error about ThreadPool not being the name of anything, find this patch as the root cause, and rename all their ThreadPool->DefaultThreadPool)

Gotcha, thanks for elaborating, somehow my brain was slow on a Monday.
Will look into this!

**Review by MaskRay — APPROVED** [review]

**Comment by joker-eph** [comment]
I did the first part of the renaming @dwblaikie : looks good?

**Comment by dwblaikie** [comment]
> I did the first part of the renaming @dwblaikie : looks good?

Hmm - this patch still seems to have both renamings in it, if I'm reading the PR correctly? I take it from the subject that isn't the intent/the intent is that his patch only does the ThreadPool->DefaultThreadPool change?

**Comment by joker-eph** [comment]
Actually no: the first patch landed already, so `ThreadPoolInterface` is now the base class in the codebase. I have some mixup here in that when renaming ThreadPool -> DefaultThreadPool, I used the base class ThreadPoolInterface when updating some of the uses.

I will push these ahead as a NFC change and rebase, that'll reduce the diff here.

**Review by dwblaikie — APPROVED** [review]

File: lldb/source/Core/Debugger.cpp [hunk 1]
```diff
@@ -104,7 +104,7 @@ static std::recursive_mutex *g_debugger_list_mutex_ptr =
     nullptr; // NOTE: intentional leak to avoid issues with C++ destructor chain
 static Debugger::DebuggerList *g_debugger_list_ptr =
     nullptr; // NOTE: intentional leak to avoid issues with C++ destructor chain
-static llvm::ThreadPool *g_thread_pool = nullptr;
+static llvm::DefaultThreadPoolThreadPool *g_thread_pool = nullptr;
```
Comment (joker-eph) [inline lldb/source/Core/Debugger.cpp:107]: Weirdly this issue wasn't caught in CI!

**Pull request merged**